src/intelligence_layer/connectors/document_index/document_index.py

```diff
@@ -29,11 +29,20 @@ class IndexConfiguration(BaseModel):
 
     Args:
         embedding_type: "symmetric" or "asymmetric" embedding type.
+        chunk_overlap: The maximum number of tokens of overlap between consecutive chunks. Must be
+            less than `chunk_size`.
```

Comment:
Question for understanding: If you have the overlap more than half of the chunk size, you will have all information doubled right?

Reply:
This is likely, but not guaranteed: the overlap is an upper bound, so there's no guarantee that your overlap will _always_ be the specified number of tokens. We do semantic chunking, so if there's a sensible semantic boundary close to the given overlap, the chunk will stop there.